### PR TITLE
added filterArray to CSX

### LINKED ISSUE #25: Please add `filterArray` to the CSX directory
As above ^^.  Remember to use the linking syntax in your PR to auto-close this issue.

### PR DESCRIPTION
**Pull request by a-hend**
Resolves #25 

Added a solution for filterArray to CSX.

### CODE REVIEW

File: Users/abehenderson/CSX/filterArray.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Higher Order Function
+function filterArray(array, callback) {
+    const newArray = [];
+    for (let i = 0; i < array.length; i += 1) {
+      if (callback(array[i])) newArray.push(array[i]);
```
Comment (C-STYR) [inline Users/abehenderson/CSX/filterArray.js:5]: This line is fine; as your experience increases you might see it written as:
```
    callback(array[i]) && newArray.push(array[i]);
```

**Review by C-STYR — APPROVED** [review]

**Comment by C-STYR** [comment]
Not sure you've properly linked to the issue, however.  Please check your syntax.

**Comment by a-hend** [comment]
Ok, I fixed it...I think.  I just deleted 'Issue', but it looks like it won't link retroactively...?

Do you want me to merge it manually?

File: Users/abehenderson/CSX/filterArray.js (hunk 1 above)
Comment (a-hend) [reply Users/abehenderson/CSX/filterArray.js:5]: Ok thanks!

Comment (a-hend) [reply Users/abehenderson/CSX/filterArray.js:5]: will that work for `return` ?

For instance, it doesn't work to do `(num % 2 === 0) ? return num` or even `(num % 2 === 0) ? return num : return null;` or `(num % 2 === 0) ? return num : 0;`

But can you use the `&&` to return `num`?  `(num % 2 === 0) && return num;`

Actually, I realize I'm asking you here, but I'll just go figure it out on my own :-)

**Comment by C-STYR** [comment]
Yes, it will link as long as it hasn't been merged.  You can go ahead and merge this one and any other PRs that have been approved.

**Pull request merged**